src/wallet-adapters/ledger_adapter.tsx

```diff
@@ -0,0 +1,85 @@
+// some code from https://github.com/project-serum/spl-token-wallet/blob/master/src/utils/walletProvider/ledger.js
+// @TODO: check license for ^
+
+import type Transport from "@ledgerhq/hw-transport";
+import type { Transaction } from "@solana/web3.js";
+
+import EventEmitter from "eventemitter3";
+import { PublicKey } from "@solana/web3.js";
+import TransportWebUSB from "@ledgerhq/hw-transport-webusb";
```

Comment:
did you check whats better webusb or webhid

Reply:
Yeah, WebHID is the newer protocol but it's hidden behind an experimental flag in Chrome per https://github.com/LedgerHQ/ledgerjs#ledgerhqhw-transport-

IMO this makes WebUSB the most widely practical choice for now, but eventually supporting (or at least providing a basic reference implementation for) WebHID, WebUSB, and Bluetooth would be ideal.

Reply:
👍